tests/test_schema.py

```diff
@@ -3027,3 +3028,27 @@ class Model(A):
             'level': 3,
         }
     }
+
+
+def test_model_with_type_attributes():
+    class Foo(BaseModel):
+        b: float
+
+    class Bar(BaseModel):
+        a: int
+        b: Type[Foo]
+
+    assert Bar.schema() == {
+        'title': 'Bar',
+        'type': 'object',
+        'properties': {'a': {'title': 'A', 'type': 'integer'}, 'b': {'$ref': '#/definitions/Foo'}},
+        'required': ['a', 'b'],
+        'definitions': {
```

Comment:
This is not correct, this is saying `b` should be an instance of `Foo`, but the type says "`b` should be `Foo` or a subclass of `Foo`".

I assume there's no equivalent in JSONSchema since you can't encode "the type `Foo`" in JSON, so I assume we'll need to use a generic "any" or similar.